### DISCUSSION TITLE
Welcome to homeassistant-mixergy Discussions!

### DISCUSSION THREAD

**Discussion by tomasmcguinness** [tomasmcguinness/homeassistant-mixergy#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by DirkFace** [comment]
Hi, I'm Dirk.
Looking forward to seeing how people are using their mixergy tank alongside HA.
It's a been a massive help for me as I have an Eddi to suck up any surplus solar into the Mixergy tank, but the smart side of the mixergy tech doesn't seem bothered in supporting any other product other than their own - to the point it doesn't even register when the water is being heated from solar on the app!

Which is why I hardly ever use the mixergy app & do everything I need to through Home assistant.
I know instantly when any surplus is being fed to the tank. I can see the water level & temperatures, then actively make a decision via an automation on wether I want to continue with Eddi heating up the water, or telling it to STOP & utilise the surplus in another area of my home.

I also have a calendar set up in HA so it can check when the tank is going to attempt an overnight cleanse, & if it's sunny I can tell Eddi to not stop until 100% of the water is heated (if surplus is available) - therfore saving me money as the Mixergy tank will then not bother doing a cleanse from the grid or gas as it will have registered that the minumum temperature has already been acheived.

There are other automations I have too, & I am also tinkering with a water usage sensor in YAML which I shall add onto here shortly when I get a few spare minutes.

**Comment by Lazulent** [comment]
Hi, I have a Mixergy iHP and a Sunsynk PV + battery.
I have used the HA Integration to recharge the tank during the day when the batteries are fully charged and surplus power is available.  This is to provide extra hot water not to save money.

At present the HA Integration does not provide tank power consumption although it is available in the REST API.  I have written my own Python program to retrieve this data.  It shows that there is an operational problem.  Normally the Heat Pump uses around 1kW and charges to the set temperature.  However, sometimes the tank consumes 2kW and charges to the 70C max.  This aberation is rectified by a power cycle suggesting it is software related.  

Including tank power consumption in the integration would be very helpful.
